compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/SkikoUIView.kt

```diff
@@ -16,86 +16,60 @@
-import androidx.compose.ui.platform.IOSSkikoInput
-import androidx.compose.ui.platform.SkikoUITextInputTraits
-import androidx.compose.ui.platform.TextActions
 import kotlinx.cinterop.*
-import org.jetbrains.skia.Rect
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skiko.SkikoInputModifiers
+import org.jetbrains.skiko.SkikoKey
+import org.jetbrains.skiko.SkikoKeyboardEvent
+import org.jetbrains.skiko.SkikoKeyboardEventKind
 import platform.CoreGraphics.*
 import platform.Foundation.*
 import platform.Metal.MTLCreateSystemDefaultDevice
 import platform.Metal.MTLDeviceProtocol
 import platform.Metal.MTLPixelFormatBGRA8Unorm
 import platform.QuartzCore.CAMetalLayer
 import platform.UIKit.*
-import platform.darwin.NSInteger
-import org.jetbrains.skia.Surface
-import org.jetbrains.skia.Canvas
-import org.jetbrains.skiko.SkikoInputModifiers
-import org.jetbrains.skiko.SkikoKey
-import org.jetbrains.skiko.SkikoKeyboardEvent
-import org.jetbrains.skiko.SkikoKeyboardEventKind
-import org.jetbrains.skiko.SkikoPointer
-import org.jetbrains.skiko.SkikoPointerDevice
-import org.jetbrains.skiko.SkikoPointerEvent
-import org.jetbrains.skiko.SkikoPointerEventKind
-
-
-internal interface SkikoUIViewDelegate {
-    fun onKeyboardEvent(event: SkikoKeyboardEvent)
-
-    fun pointInside(point: CValue<CGPoint>, event: UIEvent?): Boolean
 
-    fun onTouchesEvent(view: UIView, event: UIEvent, phase: UITouchesEventPhase)
-
-    fun retrieveInteropTransaction(): UIKitInteropTransaction
-
-    fun render(canvas: Canvas, targetTimestamp: NSTimeInterval)
-
-    /**
-     * A callback invoked when [UIView.didMoveToWindow] receives non null window
-     */
-    fun onAttachedToWindow() {}
-}
-
-internal enum class UITouchesEventPhase {
-    BEGAN, MOVED, ENDED, CANCELLED
-}
+internal class SkikoUIView(
+    private val focusable: Boolean,
+    private val keyboardEventHandler: KeyboardEventHandler,
+    private val delegate: SkikoUIViewDelegate,
+    private val transparentBackground: Boolean,
+) : UIView(
+    frame = CGRectMake(
+        x = 0.0,
+        y = 0.0,
+        width = 1.0, // TODO: Non-zero size need to first render with ComposeSceneLayer
+        height = 1.0
+    )
+) {
 
-@Suppress("CONFLICTING_OVERLOADS")
-internal class SkikoUIView : UIView, UIKeyInputProtocol, UITextInputProtocol {
     companion object : UIViewMeta() {
         override fun layerClass() = CAMetalLayer
     }
 
-    @Suppress("UNUSED") // required for Objective-C
-    @OverrideInit
-    constructor(coder: NSCoder) : super(coder) {
-        throw UnsupportedOperationException("init(coder: NSCoder) is not supported for SkikoUIView")
-    }
-
-    var delegate: SkikoUIViewDelegate? = null
-    var input: IOSSkikoInput? = null
-    var inputTraits: SkikoUITextInputTraits = object : SkikoUITextInputTraits {}
+    var onAttachedToWindow: (() -> Unit)? = null
+    private val _isReadyToShowContent: MutableState<Boolean> = mutableStateOf(false)
```

Comment:
As discussed - it's better to get rid of compose state primitives inside such classes

Reply:
For now it is better to leave it here